### PR TITLE
nixos/unifi-poller: add unifi-poller service

### LINKED ISSUE #88846: Replace unifi_exporter with unifi-poller (prometheus exporter)
**Project description**
*Collect ALL UniFi Controller, Site, Device & Client Data*

Nixpkg currently has [unifi_exporter](https://github.com/mdlayher/unifi_exporter) which is not developed anymore.

Corresponding NixOS module may have to be updated.

**Metadata**

* homepage URL: https://github.com/unifi-poller/unifi-poller/wiki
* source URL: https://github.com/unifi-poller/unifi-poller
* license: MIT
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Replace unifi_exporter and resolve #88846

It is not a drop in replacement so I added it in addition to the old unifi exporter.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/monitoring/unifi-poller.nix [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        type = types.str;
+        description = "URL of the Unifi controller.";
+        example = "https://127.0.0.1:8443";
+      };
+      username = mkOption {
+        type = types.str;
+        description = "Username used to access the unifi controller. The user needs read access.";
+        example = "unifipoller";
+      };
+      passwordFile = mkOption {
```
[the remark this answers is not in the document]
Comment (bachp) [reply nixos/modules/services/monitoring/unifi-poller.nix:25]: What are the alternatives?

File: nixos/modules/services/monitoring/unifi-poller.nix [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  prometheusEnvOptions = ''
+    # Prometheus
+    export UP_PROMETHEUS_DISABLE="${if cfg.prometheus.enable then "false" else "true"}"
+  '';
+
+in {
+  options = {
+    services.unifi-poller = {
+      enable = mkEnableOption "Enable the unifi-poller service";
```
[the remark this answers is not in the document]
Comment (bachp) [reply nixos/modules/services/monitoring/unifi-poller.nix]: Fixed

File: nixos/modules/services/monitoring/unifi-poller.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/monitoring/unifi-poller.nix:25]: Ideally using [LoadCredential](https://github.com/systemd/systemd/pull/16568). Until that lands in NixOS the better option would be a script in `ExecStart` which is executed as `root` and copies `passwordFile` into the private `WorkingDirectory` folder so the `DynamicUser` service could read it (but no one else). As a last resort you could avoid `DynamicUser`... but then you are foregoing certain security features in favour of a secure `passwordFile` option which is not ideal.

**Comment by bachp** [comment]
@lheckemann I don't think it's missing anything.

Closing in favor of #96830

**Pull request closed without merging**